turbopack/crates/turbo-tasks-backend/src/backend/mod.rs

```diff
@@ -1692,14 +1691,25 @@ impl<B: BackingStorage> TurboTasksBackendInner<B> {
         let mut old_edges = Vec::new();
 
         let has_children = !new_children.is_empty();
-        let has_mutable_children =
-            has_children && new_children.values().any(|is_immutable| !*is_immutable);
-
-        // If the task is not stateful and has no mutable children, it does not have a way to be
-        // invalidated and we can mark it as immutable.
-        if !stateful && !has_mutable_children {
-            task.mark_as_immutable();
-        }
+        let is_immutable = task.is_immutable();
+        let task_dependencies_for_late_immutable =
+            // Task is already statically immutable or
+            // previously marked as immutable
+            if !is_immutable
```

Comment:
```suggestion
        let task_dependencies_for_late_immutable =
            // Task was previously marked as immutable
            if !is_immutable
```

remove reference to the static checks?